src/btree/bt_curnext.c

```diff
@@ -691,6 +691,8 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
 			__wt_page_evict_soon(session, cbt->ref);
 		cbt->page_deleted_count = 0;
 
+		if (F_ISSET(cbt, WT_CBT_READ_WONT_NEED))
```

Comment:
@dgottlieb I had another review thought on this. I think it would be good to name the CBT flag `WT_CBT_READ_ONCE` both to keep it similar to the API naming and to avoid the similarity with the cache flag so it is visually more obvious.